nextflow.config

```diff
@@ -386,47 +370,50 @@ dag {
-includeConfig 'conf/modules.config'
+// Nextflow plugins
+plugins {
+    id 'nf-schema@2.1.2' // Validation of pipeline parameters and creation of an input channel from a sample sheet
+}
 
-// Function to ensure that resource requirements don't go beyond
-// a maximum limit
-def check_max(obj, type) {
-    if (type == 'memory') {
-        try {
-            if (obj.compareTo(params.max_memory as nextflow.util.MemoryUnit) == 1)
-                return params.max_memory as nextflow.util.MemoryUnit
-            else
-                return obj
-        } catch (all) {
-            println "   ### ERROR ###   Max memory '${params.max_memory}' is not valid! Using default value: $obj"
-            return obj
-        }
-    } else if (type == 'time') {
-        try {
-            if (obj.compareTo(params.max_time as nextflow.util.Duration) == 1)
-                return params.max_time as nextflow.util.Duration
-            else
-                return obj
-        } catch (all) {
-            println "   ### ERROR ###   Max time '${params.max_time}' is not valid! Using default value: $obj"
-            return obj
-        }
-    } else if (type == 'cpus') {
-        try {
-            return Math.min( obj, params.max_cpus as int )
-        } catch (all) {
-            println "   ### ERROR ###   Max cpus '${params.max_cpus}' is not valid! Using default value: $obj"
-            return obj
-        }
+validation {
+    defaultIgnoreParams = ["genomes"]
+    help {
+        enabled = true
+        command = "nextflow run $manifest.name -profile <docker/singularity/.../institute> --input samplesheet.csv --outdir <OUTDIR>"
+        fullParameter = "help_full"
+        showHiddenParameter = "show_hidden"
+        beforeText = """
+-\033[2m----------------------------------------------------\033[0m-
+                                        \033[0;32m,--.\033[0;30m/\033[0;32m,-.\033[0m
+\033[0;34m        ___     __   __   __   ___     \033[0;32m/,-._.--~\'\033[0m
+\033[0;34m  |\\ | |__  __ /  ` /  \\ |__) |__         \033[0;33m}  {\033[0m
+\033[0;34m  | \\| |       \\__, \\__/ |  \\ |___     \033[0;32m\\`-._,-`-,\033[0m
+                                        \033[0;32m`._,._,\'\033[0m
+\033[0;35m  ${manifest.name} ${manifest.version}\033[0m
+-\033[2m----------------------------------------------------\033[0m-
+"""
+        afterText = """${manifest.doi ? "* The pipeline\n" : ""}${manifest.doi.tokenize(",").collect { "  https://doi.org/${it.trim().replace('https://doi.org/','')}"}.join("\n")}${manifest.doi ? "\n" : ""}
+* The nf-core framework
+    https://doi.org/10.1038/s41587-020-0439-x
+
+* Software dependencies
+    https://github.com/${manifest.name}/blob/master/CITATIONS.md
+"""
+    }
+    summary {
+        beforeText = validation.help.beforeText
+        afterText = validation.help.afterText
     }
 }
+
+// Load modules.config for DSL2 module specific options
+includeConfig 'conf/modules.config'
```

Comment:
You might have this issue in `test_nopreprocessing`, `test_nothing`, and `test_prinseqplusplus`.

Reply:
But it would only happen if you put for `MALT_RUN`.
If you run `nextflow run main.nf -profile singularity,test_nopreprocessing --outdir results` then the `ext.args = { "-m ${params.malt_mode} -J-Xmx12G" }` (provided in `test.nopreprocessing`) will be overide by `ext.args = { "${meta.db_params} -m ${params.malt_mode}" }` written in `modules.config`.
I would therefore recommend you to put `includeConfig 'conf/modules.config'` just after `includeConfig 'conf/base.config'`.

Reply:
I am not familiar with this part. I saw `sarek` is doing the same as `taxprofiler`.  @jfy133 What do you think?

Reply:
It is something that as been introduce with the new template and I guess they didn't saw this kind of edge case.

Reply:
I will update it them.

Reply:
That's a good point! Have you made a PR to the template @LouisLeNezet ?

Reply:
This is done now https://github.com/nf-core/tools/pull/3301

Reply:
I've approved the PR as this modification will be made with the next template update.